### ISSUE TITLE
get window sIze and what monitor is it in (scaling related)

### ISSUE THREAD

**Issue by s77rt** [fyne-io/fyne#2079]
### Is your feature request related to a problem? Please describe:
It's related to the fact the app is too big on most monitors (with scaling: 100%)
I had to set a lower value for scaling manually, which is quite a pain for some users (the app may seem broken just because it's too big)

### Is it possible to construct a solution with the existing API?
No, I don't think so, but maybe with the help of getting the window size and the monitor it's in we can calculate a custom scaling value to **make the app at least fit in**

### Describe the solution you'd like to see:
by default when your run this app: https://github.com/s77rt/hashcat.launcher/releases/tag/v0.5.0
it will run with the default scaling 1.00 (100%) (especially for users that does not have any other fyne apps)
and the app just looks so huge and it's **not usable**
a possible solution is that at run time the app will change the scaling factor for a suitable value

Pros: It makes the app clear and visible enough to be used
Cons: the scaling factor is global, so it will affect other fyne apps too

Any ideas how to work around that?

**Comment by Jacalz** [comment]
It should scale the see interface automatically as far as I know. However, I am not so sure if it takes screen size into account (on Linux, that is). I have noted this in the past, that applications on two differently sized screens with the same resolution will look (in my opinion) way too big on the smaller screen.

**Comment by andydotxyz** [comment]
The default scale matches macOS and Windows default text size.
This is based on 120DPI - we will not scale lower than 100% unless instructed by a user.
This means if you have really low resolution monitors fonts may look large because we wanted to avoid blurry text.

You have not filled in the info about your OS, so I cannot guess. However if it is Linux then we have some additional code that tries to mimic what macOS and Windows have built in - that apps have a consistent size across screens. So if you drag from lowDPI to HiDPI the window will not double in size, it will scale appropriately. That way a user specified scale will apply universally as well.

Apps cannot change the scale factor, as this is designed to combine OS settings and user preferences.
Using fyne_settings or setting FYNE_SCALE in your environment is not a workaround, it is how you can set your user preference.

When running your app locally the font size looks correct. The window is strangely huge - but that seems to be set in the code.
<img width="1392" alt="Screenshot 2021-03-12 at 12 40 53" src="https://user-images.githubusercontent.com/294436/110941405-3cffd480-8330-11eb-9aad-9c3c48494c9c.png">

**Comment by andydotxyz** [comment]
Frrom your description it sounds more like the question is "How can I fit large content onto a small screen" which is something we need to consider all the time.
In that case the answer is careful design of layouts, it is not to apply scaling factors so everything is smaller then it will fit.

Also possible is that your OS combination (again, no details provided) is incorrectly detecting size/DPI of your hardware and if so we should fix that bug.

**Comment by s77rt** [comment]
> Frrom your description it sounds more like the question is "How can I fit large content onto a small screen" which is something > we need to consider all the time.
> In that case the answer is **careful design of layouts, it is not to apply scaling factors so everything is smaller then it will fit**.
Thanks a lot,
I based my initial design on a lower scale value (and it was kinda good even with 1.0 scale (fyne version 1.4))
After the upgrade i noticed the a diff in the size, i had to lower the scaling again and the whole app was based on a scaling different than 1 (less than 1), while per default users will have a scaling of 1 and thus the app will look huge for them unless they use the scaling value that i was using (the app was based in)
I will reconsider my app design to fit nicely in a scale of 1.00.

Closing this for now.

PS: I used a template (Feature Request template) as my initial request was about a feature request, and the template does not ask for OS info (like the Bug template does) thus the missing info.
